puzzle_generator/bytes_utils.py

```diff
@@ -5,19 +5,23 @@
 
 
 def int_to_bytes(in_value: int) -> bytes:
-    assert in_value >= 0  # nosec B101
-    number_of_bytes = in_value.bit_length() // 8
-    if 8 * number_of_bytes < in_value.bit_length():
-        number_of_bytes += 1
-    assert number_of_bytes <= 256  # nosec B101
-    assert (
-        number_of_bytes - 1 <= in_value.bit_length() // 8 <= number_of_bytes
-    )  # nosec B101
+    if in_value < 0:
+        raise ValueError("in_value must be non-negative")
+    number_of_bytes = get_num_bytes(in_value)
+    if number_of_bytes > 255:
+        raise ValueError("in_value must be 255 bytes or less")
     return number_of_bytes.to_bytes(length=1, byteorder=BYTEORDER) + in_value.to_bytes(
         length=number_of_bytes, byteorder=BYTEORDER
     )
 
 
+def get_num_bytes(in_value):
+    number_of_bytes = in_value.bit_length() // 8
```

Comment:
```suggestion
    if in_value < 0:
        raise ValueError("in_value must be non-negative")
    number_of_bytes = in_value.bit_length() // 8
```

Please add a test like `test_byte_length_raises_when_input_is_negative`.